src/de/error.rs

```diff
@@ -7,7 +7,7 @@ use serde::de;
 /// Possible errors that can arise during decoding.
 #[derive(Debug)]
 #[non_exhaustive]
-pub enum DecoderError {
+pub enum Error {
```

Comment:
Since we have a separate error type in the `de` module than at the top level (as opposed to something like `serde_json`, which has only one error type in the entire crate`, I think we still might want to have a prefix in the name of this type (and for the Result type below); it's a bit friendlier to users who need to import multiple Error types from our crate if we give them unique names.

Reply:
@patrickfreed is there precedence in other serde libraries for this naming style when there's more than one error/result defined? I agree that it's a little less user friendly to use something like `crate::de::Error` rather than `DeserializerError`.

Reply:
I think it's pretty standard to rely on the module path for these kinds of things. All the serialization libraries that I can find which have separate errors for serialization / deserialization simply name each `Error` ([`toml`](https://docs.rs/toml/0.5.6/toml/de/index.html), [`xml`](https://docs.rs/xml-rs/0.8.3/xml/reader/index.html), [`serde_urlencoded`](https://docs.rs/serde_urlencoded/0.6.1/serde_urlencoded/de/struct.Error.html)), and `serde` itself also follows this pattern with its [`ser::Error`](https://docs.rs/serde/1.0.110/serde/ser/trait.Error.html) and [`de::Error`](https://docs.rs/serde/1.0.110/serde/de/trait.Error.html) traits. I think we should keep to this pattern for consistency with other libraries.

Reply:
That makes sense, I think consistency with `serde` would be good. @saghm what do you think?

Reply:
I guess it makes sense to be consistent with our `oid::Error`. I'm not sure what you're referring to with the `xml-rs` crate, though; as it doesn't seem to have `ser` or `de` modules; do you mean [`serde_xml`](https://docs.rs/serde_xml/)?

Reply:
The `xml` crate I was referring to appears to actually just be a parsing library, not a serialization one. It has `reader` and `writer` modules with their own `Error`s which are somewhat analogous to `de` and `ser`. `serde_xml` appears to be the actual corresponding serialization crate for xml.